### PR TITLE
Add support for @augments and @mixes in storybook

### PR DESCRIPTION
**Pull request by vkalpias**
This PR changes each storybook component to show grouped Properties, Methods and Events.

It also includes data from parent classes (using the @augments keyword) and interfaces (using the @mixes keyword).

image

### CODE REVIEW

File: src/components/InfoBox/index.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const CLASS_INFOBOX = 'pcui-infobox';
 /**
  * @name InfoBox
  * @classdesc Represents an information box.
+ * @augments Container
```
Comment (willeastcott) [inline src/components/InfoBox/index.js:10]: But according to the code, it augments `BaseComponent`, not `Container`.

**Review by willeastcott — APPROVED** [review]

**Review by ellthompson — APPROVED** [review]

File: src/components/Panel/index.js [hunk 2]
```diff
@@ -32,21 +32,20 @@ const CLASS_PANEL_REMOVE = CLASS_PANEL + '-remove';
...
  * @property {boolean} sortable Gets / sets whether the panel can be reordered
- * @property {boolean} collapsed Gets / sets whether the panel is collapsed or expanded. Defaults to false.
  * @property {boolean} collapseHorizontally Gets / sets whether the panel collapses horizontally - this would be the case for side panels. Defaults to false.
  * @property {boolean} removable Gets / sets whether the panel can be removed
- * @property {number} headerSize The height of the header in pixels. Defaults to 32.
+ * @property {number} headerSize=32 The height of the header in pixels. Defaults to 32.
  * @property {string} headerText The header text of the panel. Defaults to the empty string.
- * @property {Container} header Gets the header conttainer.
+ * @property {Container} header Gets the header container.
  * @property {Container} content Gets the content conttainer.
```
Comment (ellthompson) [inline src/components/Panel/index.js:41]: Another typo here.

File: .storybook/utils/docscript.js [hunk 3]
```diff
@@ -1,70 +1,236 @@
...
+            }
+
+            entry.events[data.name] = data;
+        }
+    });
+
+    return entry;
 }
 
+export function getStoryBookDocs(data) {
```
Comment (ellthompson) [inline .storybook/utils/docscript.js:128]: Can this be renamed getStorybookDocs to match the official name?

File: src/components/Panel/index.js (hunk 2 above)
Comment (ellthompson) [inline src/components/Panel/index.js:41]: Typo here too.

File: .storybook/utils/docscript.js [hunk 4]
```diff
@@ -1,70 +1,236 @@
...
+        var typeAndDefault = convertTypeAndValue(prop.type, prop.defaultValue);
+        result[`${name}`] = {
+            name: name,
+            description: prop.description,
+            defaultValue: typeAndDefault[1],
+            control: {
+                type: typeAndDefault[0]
+            },
+            table: {
+                category: 'Properties',
```
Comment (ellthompson) [inline .storybook/utils/docscript.js:166]: Would it be possible to also use the subcategory key for inherited properties? (To include the `Inherits: Foobar` headers).

**Review by ellthompson — APPROVED** [review]
Approving with a couple of small comments.

File: .storybook/utils/docscript.js (hunk 3 above)
Comment (vkalpias) [reply .storybook/utils/docscript.js]: Done

File: .storybook/utils/docscript.js (hunk 4 above)
Comment (vkalpias) [reply .storybook/utils/docscript.js:166]: I tried various approaches with this before ending up with the current one. I think now that I found a way to do functions and events without using another category for them, we can do what you're suggesting. However I would like to do that a bit later - the current implementation shows all of the properties / functions / events even for inherited classes but just doesn't show which are inherited and which are not.

File: src/components/Panel/index.js (hunk 2 above)
Comment (vkalpias) [reply src/components/Panel/index.js]: Fixed

**Pull request merged**